### PR TITLE
New serverless pattern - awstransfer-lambda-sam

### PR DESCRIPTION
**Pull request by jackdwayne**
This pattern in SAM offers an example to setup an AWS SFTP server using AWS Transfer Family for SFTP with a custom workflow step to an AWS Lambda Function. The template enables you to simply create a server, a custom workflow, set up user accounts, and an Amazon Simple Storage Service (Amazon S3) bucket. You have fine-grained control over user identity, permissions, and keys. You can create users within Transfer for SFTP. You can also use IAM policies to control the level of access granted to each user. When a user uploads a file to the SFTP server it triggers a custom document processing step.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: transfer-lambda/example-pattern.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+  "title": "AWS Transfer Family SFTP Server to AWS Lambda",
+  "description": "This pattern in SAM offers an example to setup an AWS SFTP server using AWS Transfer Family for SFTP with a custom workflow step to an AWS Lambda Function.",
```
Comment (parikhudit) [inline transfer-lambda/example-pattern.json:3]: Length needs to be less than 150 chars.

File: transfer-lambda/example-pattern.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  "language": "Python",
+  "level": "200",
+  "framework": "SAM",
+  "introBox": {
+    "headline": "Create a AWS Transfer Family SFTP Server to custom workflow step Lambda function.",
```
Comment (parikhudit) [inline transfer-lambda/example-pattern.json:8]: Length needs to be less than 75 chars.

**Comment by parikhudit** [comment]
For a pattern that uses SAM to deploy solution for S3 to Lambda to DynamoDB, I'd suggest naming it as ```s3-lambda-dynamodb-sam```. For this case, I'd suggest renaming the folder name from ```transfer-lambda``` to something like ```awstransfer-lambda-sam```.

**Comment by jackdwayne** [comment]
Suggested changes have been made to folder and naming conventions in the README and example-pattern.json

**Pull request merged**